### PR TITLE
Minor lasgun rework, removes redundant fire mode and adds a new one

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Removes pulse mode as it was entirely redundant and was just better than regular in every way.
Adds a disabler mode that shoots disablers that slow and deal stamloss to mobs.

## Why It's Good For The Game
Removes redundant fire mode and replaces it with a cool disabler fire mode for support.

## Changelog
:cl:
add: Added disabler mode to lasgun
del: Removed pulse mode from lasgun.
balance: removed redundant pulse mode from lasgun, made regular mode fire and automatic and as fast as pulse mode, gave lasgun a new fire mode and type. Disabler shots that slow benos.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -53. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Hughgent** [comment]
If you are making this gun always automatic, and remember this affects the special firemodes as well, then remove the part of the function that sets the automatic firemode.

**Comment by Pariah919** [comment]
> If you are making this gun always automatic, and remember this affects the special firemodes as well, then remove the part of the function that sets the automatic firemode.

I didn't change the base lasgun firemode which everything inherits from, just make the regular and disabler mode automatic

File: code/modules/projectiles/guns/energy.dm [hunk 1]
```diff
@@ -490,17 +490,18 @@
 /datum/lasrifle/base/standard
 	charge_cost = 20
 	ammo = /datum/ammo/energy/lasgun/M43
-	fire_delay = 3
+	fire_delay = 2
 	fire_sound = 'sound/weapons/guns/fire/laser3.ogg'
 	message_to_user = "You set the Lasrifle's charge mode to standard fire."
-	icon_state = "tx73"
+	fire_mode = GUN_FIREMODE_AUTOMATIC
+	icon_state = "tx73_auto"
```
Comment (Hughgent) [inline code/modules/projectiles/guns/energy.dm:497]: The two modes being both the same sprite is bad. either make a new one, or make a choice as to which one should be which.

**Review by Hughgent — CHANGES_REQUESTED** [review]
sprite needs to be different per mode.

File: code/modules/projectiles/guns/energy.dm (hunk 1 above)
Comment (Pariah919) [reply code/modules/projectiles/guns/energy.dm]: done

**Review by Hughgent — APPROVED** [review]

**Comment by Surrealaser** [comment]
Honestly the lasgun is now bloated as fuck.

It should be two separate weapons: a med-long proper las rifle with standard, full auto (pulse) and overcharge, and a microwave gun that's basically an energy shotgun with heat wave and spread.

In compensation for the loss of flexibility, they can be made somewhat better at their now much more limited functions.

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Fix CL

Also lasgun is supposed to have lots of different modes thats its whole niche

**Comment by Pariah919** [comment]
@TiviPlus oops, i fixed the cl

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-64). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**